File: hw06/models/animals/birds/Bird.java
package hw06.models.animals.birds;

import hw06.models.animals.base.Animal;

/**
 * Animal is ab abstract class for birds
 */
public abstract class Bird extends Animal{
    private String feathersColor;

    /**
     * It is a constructor for Bird class
     * @param height        The height of bird
     * @param weight        The weight of bird
     * @param eyeColor      The eye color of bird
     * @param feathersColor The feathers color of bird
     */
    protected Bird(double height, double weight, String eyeColor, String feathersColor) {
        super(height, weight, eyeColor);
        this.feathersColor = feathersColor;
    }

    /**
     * Returns feathers color
     * @return this.feathersColor
     */
    public String getFeathersColor() {
        return this.feathersColor;
    }
}

File: hw07/internal/models/bird.go
package models

import "fmt"

// Bird base structure
type Bird struct {
	Animal
	feathersColor string
}

// Creates new Bird instance with full parameters given
func newBird(height float64, weight float64, eyeColor string, feathersColor string) (*Bird, error) {
	a, err := newAnimal(height, weight, eyeColor)
	if err != nil {
		return nil, err
	}

	b := &Bird{
		*a,
		feathersColor,
	}

	// Handles potential restrictions of any parameter to return error when new Bird cannot be instantiated
	return b, nil
}

// Returns the bird feathers color
func (b *Bird) FeathersColor() string {
	return b.feathersColor
}

// Returns the string representation of Animal instance
func (b *Bird) String() string {
	return fmt.Sprintf("%v, Feathers color: %s", b.Animal.String(), b.feathersColor)
}
